pyro/util.py

```diff
@@ -386,6 +387,56 @@ def check_site_shape(site, max_iarange_nesting):
     # TODO Check parallel dimensions on the left of max_iarange_nesting.
 
 
+def _are_independent(counters1, counters2):
+    for name, counter1 in counters1.items():
+        if name in counters2:
+            if counters2[name] != counter1:
+                return True
+    return False
+
+
+def check_traceenum_requirements(model_trace, guide_trace):
+    """
+    Warn if user could easily rewrite the model or guide in a way that would
+    clearly avoid invalid dependencies on enumerated variables.
+
+    :class:`~pyro.infer.traceenum_elbo.TraceEnum_ELBO` enumerates over
+    synchronized products rather than full cartesian products. Therefore models
+    must ensure that no variable outside of an iarange depends on an enumerated
+    variable inside that iarange. Since full dependency checking is impossible,
```

Comment:
... is impossible without ____ ....

Reply:
full dependency checking is impossible without *an oracle to the halting problem*